crates/bevy_ui/src/widget/image.rs

```diff
@@ -5,17 +5,19 @@ use bevy_ecs::query::Without;
 use bevy_ecs::{
     prelude::Component,
     query::With,
-    system::{Query, Res},
+    system::{Query, Res}, reflect::ReflectComponent,
 };
 use bevy_math::Vec2;
+use bevy_reflect::{Reflect, std_traits::ReflectDefault, FromReflect};
 use bevy_render::texture::Image;
 #[cfg(feature = "bevy_text")]
 use bevy_text::Text;
 
-/// The size of the image in pixels
+/// The size of the image in physical pixels
 ///
-/// This field is set automatically
-#[derive(Component, Copy, Clone, Debug, Default)]
+/// This field is set automatically by `update_image_calculated_size_system`
+#[derive(Component, Debug, Copy, Clone, Default, Reflect, FromReflect)]
+#[reflect(Component, Default)]
```

Comment:
Could we add `FromReflect` (`ReflectFromReflect`) to this list?

Reply:
oh right I'm silly I need to learn how the reflection API works properly. I've mostly just been doing what I thought was needed to support reflection, without really understanding how it worked.

So what you mean is that this should derive `Reflect`, and then reflect `FromReflect`?

Most of the components in bevy_ui only derive `Reflect` and then reflect `Component` and `Default`, is this wrong?
I thought `FromReflect` was something that you needed to derive for types with a container field like an option or a vector.